### DISCUSSION TITLE
blog/20240208/

### DISCUSSION THREAD

**Discussion by giscus-x** [xaoxuu/blog-comments#45] [Announcements]
# blog/20240208/

这是一个非常有争议的话题，自 Stellar 诞生之初，就有这个话题相关的讨论，本文将就这个话题展开详细阐述，解答一下 Stellar 为什么与众不同。

https://xaoxuu.com/blog/20240208/

**Comment by tovarsh** [comment]
非常需要，特别是记录或生活向文章

**Reply by xaoxuu** [reply]
桌面端页面本身已经很宽了，放单张图的话已经足够大了。移动端天生支持双指缩放，符合直觉。除非一行放多个图片，不过这种情况下 Stellar 默认就是支持点击放大的，例如 `gallery` 组件。

虽然双指缩放需要用到两根手指，但对用户来说，浏览不同的网页，有些网页图片支持点击放大，有些则不支持，所以点击图片的结果有不确定性，而双指缩放是系统级功能，所有设备所有网页都支持。

**Reply by tovarsh** [reply]
可选择确实是好事，但有时候习惯这一系列能放大，点开这种主题不能放大，会不习惯。

当然这是我们都使用stellar所以碰到的友人很多stellar，同主题的差异感官就明显。

扔到茫茫人海的博客中，读者就权当博主喜好了。不同主题差异感官就下降。

**Comment by BeaCox** [comment]
我现在写新内容也不开启点击放大功能了。毕竟以前看文章如果需要放大图片我都会在新标签页打开图片，放大的功能似乎是可有可无

**Reply by xaoxuu** [reply]
对啊，就算是一张需要放大才能看清的图，用户也有办法轻易实现，更何况大部分的图片都不需要放大。

**Comment by YGuang233** [comment]
非常需要。虽然很多需求设计都是因人而异的，但是大佬你提供这种可选择性的功能就是一大胜利。更别说对我这种喜欢看大图的人来说，能点击看大图对我来说有一种爽感，我看别人的文章就是东看看西看看的心态，不止看大佬分享的知识中的代码和现象，更看一些不相关的东西（喜欢用放大镜看东西），时不时的发出一些赞叹“哇，这个大佬也用这个开发工具，甚至主题和我一样的啊，哈哈哈，这是我有成为大佬的迹象啊。”时不时的引出我的推理欲望，如b站上的视频封面不允许大图，我几次就是因为看不清b站视频的图片点了进去。我甚至给我的edge浏览器装上了“浮图秀”的大图插件，这让我没点进去就解除了我封面党的心理。放大图片就是给我省去很多麻烦，什么图片不清楚，展示的看不清，放大了自己看。就我观察了两个国内的站点（含app）知乎、力扣和csdn在桌面端、手机端（含app）都是支持大图的，而app上默认双指是关闭的，进入大图后才支持双指放大图片。
在页面的布局下图片被限制在指定的大小的方块里，大的图片相对来说会小，小的图片相对来说会放大，手机上浏览器支持双指，但是电脑上并不默认支持啊。在手机上缩放页面，恢复默认的就需要再一次收缩而不是点一下（我刚才试了一下，fancybox在我手机端无法点击图片外自动关闭，右上角不显示`×`,而能正确的隐藏操作居然是往上划）。
就目前来说```stellar```的大图对我来说是必不可少的需求，总之我很是喜欢这种设计，很爽就对了，请继续保持这一功能。

**Reply by xaoxuu** [reply]
感谢回复👍，我会继续沿用按需加载的方式的。

**Comment by penndu** [comment]
如果可以的话，可以在博文中显示缩略图片，点击之后显示原图。

**Reply by xaoxuu** [reply]
这个功能一直支持的

**Reply by penndu** [reply]
我的评论工具就有问题，放大机制好像不支持markdown。

**Comment by W4J1e** [comment]
作为博主，我写文章是很不喜欢各种找图的，想着能用语言描述清楚的就用语言，整理图片有时候也比较费时间。如果要用图片，我又不希望图片只是局部的某一角，如果图片不能提供更丰富的信息（比如某软件或系统的具体配置项），那干脆用文字就够了。
作为访客，有些图片我需要放大来看，但博客不支持的时候，我会选择新标签页打开来放大了看。
所以这个功能其实重口难调，还是“我可以不用，但不能没有”吧！

**Reply by xaoxuu** [reply]
Stellar 提供的支持是最全的，前提是用 Stellar 的标签组件。

**Comment by One-CloseX** [comment]
感觉桌面端感觉不是很需要，图片如果是16:9尺寸的话，桌面端上看的很清楚，但是手机端竖屏看的时候就特别小，尤其是截图,我博客（hexo）上目前用的medium zoom，唯一的优势就是Package比较小,但是只能单纯放大，有时候放大了还缩不回去。 特地去看了下Apple Newsroom, 有些文章图片也是可以放大的，觉得还是看图片类型和尺寸。能用语言表达清楚的能不放图片就不放。还有有些截图亮色模式下截的，暗色模式下又感觉不协调，还得准备两种颜色图片根据主题切换，电脑端移动端还要准备两张图片，如果真非常在乎阅读体验得准备4张图片（如果2x清晰度还得加图片）。折腾。当然我感觉如果有这个功能还是比没有要好。

**Reply by xaoxuu** [reply]
嗯嗯是的

**Comment by songshua295** [comment]
博客真的需要图片点击放大功能吗？我觉得是必要的，博客不是txt，而是一种“富文本”，你需要通过图片、声音、动画等效果来阐述你自己的观点或者文章，既然有比文本更好的阐述方式，那有胜于无。可以不用，但不能没有。
在放代码的时候，我发现当有大片的代码片段的时候，如果是放markdown的代码块的话，实际呈现效果非常难看，因为很多对代码不感兴趣的人来说，他并不关注这个，但是却需要往下翻很久才能翻过代码块。使用图片的话，能够解决这个问题，只需要翻过一个大的图片，同时对于一些想要了解的人来说，点击放大也能帮助到想要这片代码片段的人去了解代码详情。当然，这里使用折叠的方式也能解决，但是并非是代码才需要使用图片去阐述，一些摄影师来说，分享自己的拍摄成果也是需要的。

**Reply by xaoxuu** [reply]
本文论述的观点本质是**作者能不能通过提高图片质量让读者一眼看到作者要表达的内容**，而非文章需不需要支持图片，您在评论中主要论证了图片的必要性，这与本文观点并不矛盾。
此外，尽管我认为绝大部分情况下可以做到（提高图片质量），但是我仍然为主题提供了点击放大功能。
> 在不得不贴大量代码的场景中，我更推荐以附件的形式提供，这样读者可以复制（贴图片就不可以复制了，而且占用的流量也比源码更大）。

**Comment by dosedo** [comment]
嗯呢

**Comment by cxplay** [comment]
Stellar 的 fancybox 只能在使用了 `{% image %}` 标签插件的地方开启吗?

**Reply by xaoxuu** [reply]
是的

**Comment by TrueInFalse** [comment]
mark
